engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2405,6 +2540,46 @@ private LogicalOperatorInfo fillLogicalOperatorInfo(OperatorMeta operator)
     return loi;
   }
 
+  private LogicalModuleInfo fillLogicalModuleInfo(ModuleMeta module, boolean flatten, String parentModuleName)
+  {
+    LogicalModuleInfo lmi = new LogicalModuleInfo();
+    lmi.name = module.getName();
+    lmi.className = module.getModule().getClass().getName();
+    if (flatten) {
+      for (OperatorMeta operatorMeta : getLogicalPlan().getAllOperators()) {
+        if (operatorMeta.getModuleName() != null
+            && (operatorMeta.getModuleName().equals(module.getName()) || operatorMeta.getModuleName().equals(
+                parentModuleName))) {
+          lmi.operators.add(fillLogicalOperatorInfo(operatorMeta));
+        }
+      }
+      for (StreamMeta streamMeta : getLogicalPlan().getAllStreams()) {
+        if (streamMeta.getModuleName() != null
+            && (streamMeta.getModuleName().equals(module.getName()) || streamMeta.getModuleName().equals(
+                parentModuleName))) {
+          lmi.streams.add(fillLogicalStreamInfo(streamMeta));
```

Comment:
I think @tweise pointed it out earlier as well, why do we need to iterate over streams when we have done over operators.